### ISSUE TITLE
May I continue training after some epochs are done but giving more epochs?

### ISSUE THREAD

**Issue by ciaochiaociao** [flairNLP/flair#1824]
Hello Professor Akbik,

I have trained with max_epochs set to 20 and found that it is still not reaching the plateau. So I would like to continue the training and see if it will converge and get even better performance. 

Although I did not train with the argument `checkpoint` set to `True`, I use Jupyter notebook to train my model, where variables are still there including `trainer`. So theoretically, I can just continue the training by `trainer.train(..., max_epochs=40)` to train with more epochs, am I right?

I know you documented the "resume training" part in one of your tutorial, Training the model. However, there is a slight difference. The tutorial says you first `trainer.train(..., max_epochs=150)` and then `trainer.train(..., max_epochs=150)` when you want to resume training where the `max_epochs` was set to **the same value**. 

But my case is I finished the training for all epochs and want to continue training with more epochs, where the `max_epochs` argument are set differently. In short, my question is **may I resume training by giving a different and larger `max_epochs`?**

(Note:

I know that some `scheduler`s will use `max_epochs` to pre-calculate the scheduled learning rates for each epoch like linear scheduler, but I think the default scheduler in `ModelTrainer`, AnnealOnPlateau`, does not depend on `max_epochs` but other performance criterion to determine the next learning rate. 

**So I think it is safe to continue training with different `trainer.train()` with different `max_epochs` from the previous training process given the scheduler `AnnealOnPlateau`.** 

But if a linear scheduler is used like `huggingface.transformers` do, which calculates the learning rate for each epoch according to the `max_epochs`, this is not the case.

Please correct me if I were wrong!

And by the way, congratulation to you that you have become a professor. Just saw that on your personal page. I really appreciate your elegant NLP framework, flair. I am now a big fan of this easy-to-use tool. Really thank you for your contribution to this field.

**Comment by alanakbik** [comment]
Hello @ciaochiaociao thanks a lot! and please excuse the delay in response. Yes, I think re-loading should work in this case. If the learning rate annealed in the first training run you could set a lower learning rate when you resume.

**Comment by ciaochiaociao** [comment]
Okay. Thank you.
